Rust structs 1 (#3470)
Codecademy/docs: content/rust/concepts/structs/structs.md

---
Title: 'Structs'

> Description: 'Rust structs are custom data structures used to group related data fields together. They define the blueprint for creating instances with named fields, enabling organized and type-safe data storage.'

Reviewer: ```suggestion
Description: 'Structs are custom data structures used to group related data fields together.'
```

Subjects:

>   - 'Data Modeling'

Reviewer: ```suggestion
  - 'Code Foundations'
```

  - 'Web Development'

>   - 'Blockchain and Cryptocurrency'
>   - 'Graphics and Computer Vision'

Reviewer: ```suggestion
```

Reviewer: Subjects are limited to two items and must come from the subjects.md list

Tags:
  - 'Comments'
  - 'Documentation'
CatalogContent:

>   - 'rust-for-programmers'

Reviewer: ```suggestion
  - 'rust-for-programmers'
  - 'paths/computer-science'  
```

---

> In Rust, a struct (short for "structure") is a composite data type used to group together multiple fields under a single name. Each field in a struct can have its own data type, allowing you to represent complex data structures. Structs are defined using the 'struct' keyword.

Reviewer: ```suggestion
In Rust, a **struct** (short for "structure") is a composite data type used to group together multiple fields under a single name. Each field in a struct can have its own data type, permitting the representation of complex data structures. Structs are defined using the `struct` keyword.
```

> ## Defining Struct

Reviewer: ```suggestion
## Defining a Struct
```

```rust
struct Person {
    name: Boric,
    age: 22,
    address: String,
}

> ```

Reviewer: there should always be an empty line after the code block

> In this Person struct, we have three compartments (fields): name, age, and address. Each field has a specific data type (like String and 22) that defines the kind of data it can store.

Reviewer: ```suggestion
In this `Person` struct, there are three compartments (fields): `name`, `age`, and `address`. Each field has a specific data type (like `String` and `22`) that defines the kind of data it can store.
```

## Creating Instances of Structs

> Now that we've defined our Person struct, we can create individual instances of it. These instances are like filling out the information in our 'Person' box

Reviewer: ```suggestion
Now, with the `Person` struct defined, individual instances of it can be generated. These instances are like filling out the information in our 'Person' box.
```
